src/backend/distributed/utils/connection_cache.c

```diff
@@ -30,8 +30,8 @@
 #include "utils/palloc.h"
 
 
-/* state needed to prevent new connections during modifying transactions */
-bool IsModifyingTransaction = false;
+/* state needed to keep track of operations used during a transaction */
+XactModificationType XactModification = XACT_MODIFICATION_NONE;
```

Comment:
Maybe change the variable name to something like XactModificationState to be more descriptive, or maybe *Level, since we're doing > comparisons.